clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2168,29 +2168,33 @@ Value *ScalarExprEmitter::VisitCastExpr(CastExpr *CE) {
       return EmitLoadOfLValue(DestLV, CE->getExprLoc());
     }
 
-    bool asmjs = CGF.CurFn && CGF.CurFn->getSection() == StringRef("asmjs");
-    //We don't care about casts to functions types
-    if (CGF.getTarget().isByteAddressable() || isa<llvm::ConstantPointerNull>(Src) ||
-        (isa<llvm::FunctionType>(SrcTy->getPointerElementType()) && isa<llvm::FunctionType>(DstTy->getPointerElementType())) ||
-        DstTy == SrcTy)
+    // If both are vectors, skip directly to the cast.
+    if (!(SrcTy->isVectorTy() && DstTy->isVectorTy()))
```

Comment:
This is ok, but maybe it would be nicer to avoid the extra if() around everything and make it a case of the inner if/else chain. it could be the first if() with nothing as body